pydantic/json.py

```diff
@@ -13,11 +13,22 @@ def isoformat(o):
     return o.isoformat()
 
 
+def iso8601_duration(value):
+    seconds = value.total_seconds()
+    minutes, seconds = divmod(seconds, 60)
+    hours, minutes = divmod(minutes, 60)
+    days, hours = divmod(hours, 24)
+    days, hours, minutes = map(int, (days, hours, minutes))
+    seconds = round(seconds, 6)
+    return f'P{days}DT{hours}H{minutes}M{seconds}S'
+
+
 ENCODERS_BY_TYPE = {
     UUID: str,
     datetime.datetime: isoformat,
     datetime.date: isoformat,
     datetime.time: isoformat,
+    datetime.timedelta: iso8601_duration,
```

Comment:
I guess the default should just be `lambda td: f'{td.total_seconds():0.3f}'`

Reply:
I think it's a good idea except that timedelta has a resolution of up to 1 microsecond, I would suggest formatting to `:0.6f`